frame/collective/src/lib.rs

```diff
@@ -968,6 +976,65 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		});
 		num_proposals as u32
 	}
+
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// The following expectation must always apply.
+	///
+	/// ## Expectations:
+	///
+	/// Looking at proposals:
+	///
+	/// * Each hash of a proposal that is stored inside `Proposals` must have a
+	/// call mapped to it inside the `ProposalsOf` storage map.
+	/// * `ProposalCount` must always be more or equal to the number of
+	/// proposals inside the `Proposals` storage value. The reason why
+	/// `ProposalCount` can be more is because when a proposal is removed the
+	/// count is not deducted.
```

Comment:
I think another invariant to check is that the count of values in `ProposalsOf` matches the count of values in `Proposals`. Combined with the first check, this implicitly checks that each key (hash) in `ProposalsOf` exists in `Proposals`.

Reply:
also, the `threshold` of a proposal should always be `> 1`